ocserv-country-blocking.md

```diff
@@ -0,0 +1,55 @@
+# Country Blocking
+
+Author: Lance LeFlore
+
+When providing VPN services, it is expected that clients will be connecting from a vast number of different subnets around the internet at-large. Because of this, it's not entirely feasible to use a packet filter to whitelist the subnets from which clients may be connecting to the VPN server. Very often, however, one may safely assume that clients will only be connecting from a particular set of countries. This then gives us a parameter we can in turn use to whitelist client connections. For example, if one assumes his or her clients will only ever connect from the US and Czech Republic; a GeoIP lookup can be performed against the clients' source IP to determine whether they are in fact connecting from the US or the Czech Republic.
+
+### Prerequisites
+
+- cURL
+- Tested on Ocserv 0.11.5.
+
+### Configuration
+- Set ```connect-script = /usr/bin/ocserv-country-blocker``` in ```ocserv.conf```.
+- Create a file with the following code at ```/usr/bin/ocserv-country-blocker```:
+
+``` sh
+#!/bin/sh
+
+ALLOW_COUNTRY_CODES="CZ US"
+
+# Make sure Ocserv passes IP_REAL
+if [ "x${IP_REAL}" = "x" ]; then
+  exit 1
+fi
+
+GEOIP_STRING=$(/bin/curl -s https://freegeoip.net/csv/$IP_REAL)
```

Comment:
Would it make sense to mention here in a comment the local equivalent using a tool like ipcalc? that is something like:
```
# Alternatively in fedora24 or better the following would also provide the country code variable using the local geoip database
# eval $(ipcalc -g ${IP_REAL})
```